hicommonwealth/commonwealth
# CE subscriber edge case fixes

## Link to Issue
Closes: #3992 

## Description of Changes
- Removes the use of the `runSubscriberAsFunction` from `runSubscriberAsServer`.
- `getSubscriberChainData` returns an empty array if there are no cached chains.

## Test Plan
- Run `CHAIN=dydx yarn start-subscriber` and observe logs
- Run `yarn start-subscriber` and observe logs (concerned with start-up logs not chain-specific event errors)
- Deployed to chain-events-staging and verified subscribers were running correctly

## Deployment Plan
This should be merged and deployed with the planned deployment this week. See https://github.com/hicommonwealth/commonwealth/issues/3973

## Other Considerations
